thali/thalireplicationmanager.js

```diff
@@ -38,36 +39,70 @@ ThaliReplicationManager.events = {
 };
 
 /**
-* Starts the Thali replication manager with the given device name and port number
-* @param {String} deviceName the device name to advertise.
+* Starts the Thali replication manager with the given port number and db name
 * @param {Number} port the port number used for synchronization.
+* @param {String} dbName the db name to sync.
 */
-ThaliReplicationManager.prototype.start = function (deviceName, port, dbName) {
-  validations.ensureNonNullOrEmptyString(deviceName, 'deviceName');
+ThaliReplicationManager.prototype.start = function (port, dbName) {
   validations.ensureValidPort(port);
   validations.ensureNonNullOrEmptyString(dbName, 'dbName');
 
-  this.emit(ThaliReplicationManager.events.STARTING);
-
-  this._port = port;
-  this._deviceName = deviceName;
-  this._dbName = dbName;
-  this._serverBridge = muxServerBridge.call(this, port);
-  this._serverBridge.listen(function () {
-    this._serverBridgePort = this._serverBridge.address().port;
+  var cryptomanager = require('./thalicryptomanager');
+  // get the device public-key-hash
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
+    var currentAddrEntry;
+    if (err) {
+      console.log('getPublicKeyHash returned err: ', err);
+      this._isStarted = false;
+      this.emit(ThaliReplicationManager.events.START_ERROR, err);
+      return;
+    } else {
+      console.log('got a valid publicKeyHash');
+      currentAddrEntry = addressPrefix + publicKeyHash;
```

Comment:
This should not be here per my previous comment. I think it makes sense for the cryptomanager to be here. I think it makes sense for the cryptomanager to be part of start up (after all, you can't do replication with an identity). I even think it makes sense for the cryptomanager to get the publickeyhash. But that is where it ends. That publickeyhash should be available as a property identifying the current device but that's it. It should be up to the postcardapp to then get that publickeyhash and use it for something.

Please propose a design for how we can expose the publickeyhash all the way to the postcardapp in a manner that makes logical sense.